src/cli/commands/help.js

```diff
@@ -17,24 +21,47 @@ export function run(
   const getDocsInfo = (name) => 'Visit ' + chalk.bold(getDocsLink(name)) + ' for documentation about this command.';
 
   if (args.length) {
-    const helpCommand = hyphenate(args[0]);
-    if (commands[helpCommand]) {
-      commander.on('--help', () => console.log('  ' + getDocsInfo(helpCommand) + '\n'));
-    }
-  } else {
-    commander.on('--help', () => {
-      console.log('  Commands:\n');
-      for (const name of Object.keys(commands).sort(sortAlpha)) {
-        if (commands[name].useless) {
-          continue;
+    const commandName = camelCase(args.shift());
+
+    if (commandName) {
```

Comment:
Actually I think it's done userland, more precisely [here](https://github.com/yarnpkg/yarn/pull/2887/files#diff-776aa1c3229433c4c56fe8345fd89668R52). That being said, I've checked locally and it seems that this display (one command each line) was already printed this way in recent Yarn releases, so let's keep it this way. 👍